ivy/functional/backends/numpy/activations.py

```diff
@@ -16,38 +17,54 @@ def relu(x: np.ndarray, out: Optional[np.ndarray] = None) -> np.ndarray:
     return np.maximum(x, 0, out=out)
 
 
-def leaky_relu(x: np.ndarray, alpha: Optional[float] = 0.2) -> np.ndarray:
-    return np.where(x > 0, x, x * alpha)
+def leaky_relu(
+    x: np.ndarray,
+    alpha: Optional[float] = 0.2,
+    out: Optional[np.ndarray] = None
+) -> np.ndarray:
+    ret = np.asarray(np.where(x > 0, x, x * alpha), x.dtype)
+    if ivy.exists(out):
+        return ivy.inplace_update(out, ret)
+    return ret
 
 
-def gelu(x, approximate: Optional[bool] = True):
+def gelu(x, approximate: Optional[bool] = True, out: Optional[np.ndarray] = None):
     if erf is None:
         raise Exception(
             "scipy must be installed in order to call ivy.gelu with a numpy backend."
         )
     if approximate:
-        return 0.5 * x * (1 + np.tanh(np.sqrt(2 / np.pi) * (x + 0.044715 * x**3)))
-    return 0.5 * x * (1 + erf(x / np.sqrt(2)))
+        ret = 0.5 * x * (1 + np.tanh(np.sqrt(2 / np.pi) * (x + 0.044715 * x ** 3)))
+    else:
+        ret = 0.5 * x * (1 + erf(x / np.sqrt(2)))
+    if ivy.exists(out):
+        return ivy.inplace_update(out, ret)
+    return ret
 
 
 def sigmoid(x: np.ndarray, out: Optional[np.ndarray] = None) -> np.ndarray:
-    return 1 / (1 + np.exp(-x, out=out))
+    ret = (1 / (1 + np.exp(-x)).astype(x.dtype))
+    if ivy.exists(out):
+        return ivy.inplace_update(out, ret)
+    return ret
 
 
-def tanh(x: np.ndarray, out: Optional[np.ndarray] = None) -> np.ndarray:
-    return (np.exp(x, out=out) - np.exp(-x, out=out)) / (
-        np.exp(x, out=out) + np.exp(-x, out=out)
-    )
+def tanh(
+        x: np.ndarray, out: Optional[np.ndarray] = None) -> np.ndarray:
+    ret = (np.exp(x) - np.exp(-x)) / (np.exp(x) + np.exp(-x))
+    if ivy.exists(out):
```

Comment:
we should not handle `out` in the backend. We should remove the `out` argument here unless the backend function itself is able to handle this for us.